ProcessLib/ComponentTransport/ComponentTransportFEM.h

```diff
@@ -358,27 +371,329 @@ class LocalAssemblerData : public ComponentTransportLocalAssemblerInterface
         }
     }
 
+    void assembleForStaggeredScheme(
+        double const t,
+        std::vector<double>& local_M_data,
+        std::vector<double>& local_K_data,
+        std::vector<double>& local_b_data,
+        LocalCoupledSolutions const& coupled_xs) override
+    {
+        if (coupled_xs.process_id == hydraulic_process_id)
+        {
+            assembleHydraulicEquation(t, local_M_data, local_K_data,
+                                      local_b_data, coupled_xs);
+        }
+        else
+        {
+            // Component-wise
+            assembleComponentTransportEquation(t, local_M_data, local_K_data,
+                                               local_b_data, coupled_xs);
+        }
+    }
+
+    void assembleHydraulicEquation(double const t,
+                                   std::vector<double>& local_M_data,
+                                   std::vector<double>& local_K_data,
+                                   std::vector<double>& local_b_data,
+                                   LocalCoupledSolutions const& coupled_xs)
+    {
+        auto local_p = Eigen::Map<const NodalVectorType>(
+            coupled_xs.local_coupled_xs[hydraulic_process_id].data(),
+            pressure_size);
+        int const transport_process_id = 1;
```

Comment:
Is this `transport_process_id` only used in `assembleHydraulicEquation()`-method or is this value used somewhere else?

Reply:
used in the function members where we need to consider the dependency of fluid density on the concentration:

- assembleHydraulicEquation() at Line 403;
- getIntPtDarcyVelocity() at Line 681;
- Eigen::Vector3d getFlux() at Line 777;

Reply:
Then this should not be a local variable. What if someone changes it only in one of those functions.

Reply:
Fixed. First_transport_process_id used instead.